File: problems/implement-strstr/boilerplate-full/function.java
//USER_CODE_HERE
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String haystack = sc.next();
        String needle = sc.next();
        Solution sol = new Solution();
        int result = sol.strStr(haystack, needle);
        System.out.println(result);
    }
}

File: problems/implement-strstr/boilerplate-full/function.py
#USER_CODE_HERE
import sys
from typing import List

# Set recursion limit higher for recursive problems
sys.setrecursionlimit(2000)

if __name__ == "__main__":
    # Input reading for the specific problem
    haystack = input().strip()
    needle = input().strip()

    sol = Solution()
    
    # Call the solution function
    result = sol.strStr(haystack, needle)
    
    # Print the result
    print(result)